connectors/sources/gcs.py

```diff
@@ -0,0 +1,311 @@
+                "label": "JSON string for Google cloud service account",
+                "type": "str",
+            },
+            "connector_name": {
+                "value": "Google Cloud Storage Connector",
+                "label": "Friendly name for the connector",
+                "type": "str",
+            },
+            "retry_count": {
+                "value": DEFAULT_RETRY_COUNT,
+                "label": "Retry count for failed requests",
+                "type": "int",
+            },
+            "enable_content_extraction": {
+                "value": DEFAULT_CONTENT_EXTRACTION,
+                "label": "Flag to check if content extraction is enabled or not",
+                "type": "bool",
+            },
+        }
+
+    async def _api_call(
+        self,
+        resource,
+        method,
+        sub_method=None,
+        full_response=False,
+        **kwargs,
+    ):
+        """Method for adding retries whenever exception raised during an api calls
+
+        Args:
+            resource (aiogoogle.resource.Resource): Resource name for which api call will be made.
+            method (aiogoogle.resource.Method): Method available for the resource.
+            sub_method (aiogoogle.resource.Method, optional): Sub-method available for the method. Defaults to None.
+            full_response (bool, optional): Specifies whether the response is paginated or not. Defaults to False.
+
+        Raises:
+            exception: A instance of an exception class.
+
+        Yields:
+            Dictionary: Response returned by the resource method.
+        """
+        retry_counter = 0
+        while True:
+            try:
+                async with Aiogoogle(
+                    service_account_creds=self.service_account_credentials
+                ) as google_client:
+                    storage_client = await google_client.discover(
+                        api_name=API_NAME, api_version=API_VERSION
+                    )
+                    if not sub_method and STORAGE_EMULATOR_HOST:
```

Comment:
let's add a flag to bypass any code that should not run in production when using the emulator.
Some variable exported in `ftest.sh` like `RUNNING_FTEST`